lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -263,12 +263,13 @@ def exec_module(self, **kwargs):
         # unmount from the old virtual machine and mount to the new virtual machine
         vm_name = parse_resource_id(disk_instance.get('managed_by', '')).get('name') if disk_instance else None
         if self.managed_by != vm_name:
-            changed = True
             if not self.check_mode:
                 if vm_name:
-                    self.detach(vm_name, result)
+                    if self.detach(vm_name, result):
+                        changed = True
                 if self.managed_by:
-                    self.attach(self.managed_by, result)
+                    if self.attach(self.managed_by, result):
```

Comment:
just curious, besides these 2 ifs, what else situation there could be makes changed=false?

Reply:
there's self.is_different() call that checks idempotency beforehand